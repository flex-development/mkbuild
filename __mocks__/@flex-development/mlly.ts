/**
 * @file Mocks - @flex-development/mlly
 * @module mocks/flex-development/mlly
 * @see https://github.com/flex-development/mlly
 */

/**
 * [`@flex-development/mlly`][1] module type.
 *
 * [1]: https://github.com/flex-development/mlly
 */
type Actual = typeof import('@flex-development/mlly')

/**
 * `@flex-development/mlly` module.
 *
 * @const {Actual} actual
 */
const actual: Actual = await vi.importActual<Actual>('@flex-development/mlly')

export const CONDITIONS = actual.CONDITIONS
export const RESOLVE_EXTENSIONS = actual.RESOLVE_EXTENSIONS
export const fillModules = vi.fn(actual.fillModules)
export const findStaticImports = vi.fn(actual.findStaticImports)
export const getSource = vi.fn(actual.getSource)
export const readPackageJson = vi.fn(actual.readPackageJson)
export const resolveModule = vi.fn(actual.resolveModule)
export const resolveModules = vi.fn(actual.resolveModules)
export const toDataURL = vi.fn(actual.toDataURL)
export const toURL = vi.fn(actual.toURL)
