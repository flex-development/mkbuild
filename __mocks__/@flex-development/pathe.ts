/**
 * @file Mocks - @flex-development/pathe
 * @module mocks/flex-development/pathe
 * @see https://github.com/flex-development/pathe
 */

/**
 * [`@flex-development/pathe`][1] module type.
 *
 * [1]: https://github.com/flex-development/pathe
 */
type Actual = typeof import('@flex-development/pathe')

/**
 * `@flex-development/pathe` module.
 *
 * @const {Actual} actual
 */
const actual: Actual = await vi.importActual<Actual>('@flex-development/pathe')

export const basename = vi.fn(actual.basename)
export const dirname = vi.fn(actual.dirname)
export const extname = vi.fn(actual.extname)
export const format = vi.fn(actual.format)
export const join = vi.fn(actual.join)
export const parse = vi.fn(actual.parse)
export const relative = vi.fn(actual.relative)
export const resolve = vi.fn(actual.resolve)

export default {
  basename,
  dirname,
  extname,
  format,
  join,
  parse,
  relative,
  resolve
}
