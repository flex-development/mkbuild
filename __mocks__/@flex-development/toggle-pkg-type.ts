/**
 * @file Mocks - @flex-development/toggle-pkg-type
 * @module mocks/flex-development/toggle-pkg-type
 * @see https://github.com/flex-development/toggle-pkg-type
 */

export default vi.fn()
