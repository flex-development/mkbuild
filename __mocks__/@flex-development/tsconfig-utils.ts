/**
 * @file Mocks - @flex-development/tsconfig-utils
 * @module mocks/flex-development/tsconfig-utils
 * @see https://github.com/flex-development/tsconfig-utils
 */

/**
 * [`@flex-development/tsconfig-utils`][1] module type.
 *
 * [1]: https://github.com/flex-development/tsconfig-utils
 */
type Actual = typeof import('@flex-development/tsconfig-utils')

/**
 * `@flex-development/tsconfig-utils` module.
 *
 * @const {Actual} actual
 */
const actual: Actual = await vi.importActual<Actual>(
  '@flex-development/tsconfig-utils'
)

export const ScriptTarget = actual.ScriptTarget
export const normalizeCompilerOptions = vi.fn(actual.normalizeCompilerOptions)
export const normalizeModuleResolution = vi.fn(actual.normalizeModuleResolution)
export const resolvePaths = vi.fn(actual.resolvePaths)
