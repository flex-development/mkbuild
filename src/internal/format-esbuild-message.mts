/**
 * @file Internal - formatEsbuildMessage
 * @module mkbuild/internal/formatEsbuildMessage
 */

import { Location } from '@flex-development/vfile-location'
import type { LogLevel, Message } from 'esbuild'
import type { RollupLog, LogLevel as RollupLogLevel } from 'rollup'

/**
 * Convert an esbuild `message` to a rollup log.
 *
 * @see {@linkcode LogLevel}
 * @see {@linkcode Message}
 * @see {@linkcode RollupLog}
 *
 * @internal
 *
 * @this {void}
 *
 * @param {Message} message
 *  The message to handle
 * @param {string} code
 *  The code being transformed
 * @param {string} id
 *  Module id
 * @param {string} plugin
 *  Plugin name
 * @param {LogLevel | undefined} [level='warning']
 *  esbuild log level override
 * @return {RollupLog}
 *  `message` formatted as rollup log
 */
function formatEsbuildMessage(
  this: void,
  message: Message,
  code: string,
  id: string,
  plugin: string,
  level: LogLevel | undefined = 'warning'
): RollupLog {
  /**
   * esbuild {@linkcode message} as rollup log.
   *
   * @const {RollupLog} log
   */
  const log: RollupLog = {
    hook: 'transform',
    id,
    message: message.text,
    meta: { level, notes: message.notes },
    plugin,
    pluginCode: message.id
  }

  if (level !== 'silent') {
    log.level = ({
      debug: 'debug',
      error: 'error',
      info: 'info',
      verbose: 'debug',
      warning: 'warn'
    } as Record<LogLevel, RollupLogLevel | 'error'>)[level]
  }

  if (message.location) {
    log.frame = message.location.lineText

    log.loc = {
      column: message.location.column + 1,
      file: message.location.file,
      line: message.location.line
    }

    log.pos = new Location(code).offset(log.loc)
  }

  return log
}

export default formatEsbuildMessage
