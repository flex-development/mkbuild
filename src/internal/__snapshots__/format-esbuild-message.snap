// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:internal/formatEsbuildMessage > should return \`message\` as rollup log 1`] = `
{
  "frame": "        : (await import(url.href) as { default?: Config | null }).default",
  "hook": "transform",
  "id": \${process.cwd()}/src/utils/load-build-config.mts,
  "level": "warn",
  "loc": {
    "column": 18,
    "file": \${process.cwd()}/src/utils/load-build-config.mts,
    "line": 96,
  },
  "message": "This "import" expression will not be bundled because the argument is not a string literal",
  "meta": {
    "level": "warning",
    "notes": [],
  },
  "plugin": "mkbuild:esbuild",
  "pluginCode": "unsupported-dynamic-import",
  "pos": 2183,
}
`;
